docker_image_cleaner/cleaner.py

```diff
@@ -173,83 +148,61 @@ def main():
 
     while True:
         used = get_used(path_to_check)
         logging.info(used_msg.format(used=used))
         if used < gc_high:
             # Do nothing! We have enough space
-            pass
+            time.sleep(interval)
+            continue
+
+        images = client.images.list(all=True)
+        if not images:
+            logging.info("No images to delete")
+            time.sleep(interval)
+            continue
         else:
-            images = get_docker_images(client)
-            if not images:
-                logging.info(f"No images to delete")
-                time.sleep(interval)
-                continue
-            else:
-                logging.info(f"{len(images)} images available to prune")
-
-            start = time.perf_counter()
-            images_before = len(images)
-
-            if node:
-                cordon(kube, node)
-
-            deleted = 0
-
-            while images and get_used(path_to_check) > gc_low:
-                # Ensure the node is still cordoned
-                if node:
-                    cordon(kube, node)
-                # Remove biggest image
-                image = images.pop(0)
-                if image.tags:
-                    # does it have a name, e.g. jupyter/base-notebook:12345
-                    name = image.tags[0]
-                else:
-                    # no name, use id
-                    name = image.id
-                gb = image.attrs["Size"] / (2**30)
-                logging.info(f"Removing {name} (size={gb:.2f}GB)")
+            logging.info(f"{len(images)} images available to prune")
+
+        # Ensure the node is cordoned while we prune
+        with cordon_context():
+            for kind in ("containers", "images"):
+                key = f"{kind.title()}Deleted"
+                tic = time.perf_counter()
+                collection = getattr(client, kind)  # client.containers
                 try:
-                    client.images.remove(image=image.id, force=True)
-                    logging.info(f"Removed {name}")
-                    # Delay between deletions.
-                    # A sleep here avoids monopolizing the Docker API with deletions.
-                    time.sleep(delay)
-                except docker.errors.APIError as e:
-                    if e.status_code == 409:
-                        # This means the image can not be removed right now
-                        logging.info(f"Failed to remove {name}, skipping this image")
-                        logging.info(str(e))
-                    elif e.status_code == 404:
-                        logging.info(f"{name} not found, probably already deleted")
-                    else:
-                        if node:
-                            # uncordon before giving up
-                            uncordon(kube, node)
-                        raise
+                    pruned = collection.prune()
```

Comment:
I wanted to look up and see if I felt this made sense - prune both containers and images? Yepp it did I think, and to do it in the order of containers -> images.

I'd love to have this client named `docker_client` or similar as `client` just makes me confused what this is about. In this code file, I think we have the k8s client as well.

```suggestion
                    # client.containers.prune: https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.ContainerCollection.prune
                    # client.images.prune: https://docker-py.readthedocs.io/en/stable/images.html#docker.models.images.ImageCollection.prune
                    pruned = collection.prune()
```

Reply:
docker client is now called `docker_client`